src/zkp/mod.rs

```diff
@@ -30,23 +30,23 @@ use serde::{de::DeserializeOwned, Serialize};
 ///
 /// This context can be turned into bytes and appended to a [`Transcript`].
 pub(crate) trait ProofContext {
-    fn as_bytes(&self) -> &[u8];
+    fn as_bytes(&self) -> Vec<u8>;
 }
 
 /// This is a temporary preliminary implementation that will be modified by
 /// issue #240.
 impl ProofContext for () {
-    fn as_bytes(&self) -> &[u8] {
-        &[]
+    fn as_bytes(&self) -> Vec<u8> {
+        vec![]
     }
 }
```

Comment:
note: I was going to say make this `#[cfg(test)]` now, but I tried and I think we missed a non-test-case zk proof that uses an empty context. So, I'll make a separate issue about that bug and you can handle both things there.

Reply:
Added #299.